fabric-player-tracking-api-v1/src/main/java/net/fabricmc/fabric/api/networking/player/tracking/v1/PlayerTracking.java

```diff
@@ -0,0 +1,180 @@
+	 * @return the players in the server world
+	 */
+	public static Collection<ServerPlayerEntity> world(ServerWorld world) {
+		Objects.requireNonNull(world, "The world cannot be null");
+		// return an immutable collection to guard against accidental removals.
+		return Collections.unmodifiableCollection(world.getPlayers());
+	}
+
+	/**
+	 * Gets all players tracking a chunk in a server world.
+	 *
+	 * @param world the server world
+	 * @param pos   the chunk in question
+	 * @return the players tracking the chunk
+	 */
+	public static Stream<ServerPlayerEntity> tracking(ServerWorld world, ChunkPos pos) {
+		Objects.requireNonNull(world, "The world cannot be null");
+		Objects.requireNonNull(pos, "The chunk pos cannot be null");
+		return world.getChunkManager().threadedAnvilChunkStorage.getPlayersWatchingChunk(pos, false);
+	}
+
+	/**
+	 * Gets all players tracking an entity in a server world.
+	 *
+	 * <p>The returned collection is immutable.
+	 *
+	 * <p><b>Warning</b>: If the provided entity is a player, it is not
+	 * guaranteed by the contract that said player is included in the
+	 * resulting stream.
+	 *
+	 * @param entity the entity being tracked
+	 * @return the players tracking the entity
+	 * @throws IllegalArgumentException if the entity is not in a server world
+	 */
+	public static Collection<ServerPlayerEntity> tracking(Entity entity) {
+		Objects.requireNonNull(entity, "Entity cannot be null");
+		ChunkManager manager = entity.world.getChunkManager();
+
+		if (manager instanceof ServerChunkManager) {
+			ThreadedAnvilChunkStorage storage = ((ServerChunkManager) manager).threadedAnvilChunkStorage;
+
+			// return an immutable collection to guard against accidental removals.
+			return Collections.unmodifiableCollection(((ThreadedAnvilChunkStorageTrackingExtensions) storage).fabric_getTrackingPlayers(entity));
+		}
+
+		throw new IllegalArgumentException("Only supported on server worlds!");
+	}
+
+	/**
+	 * Gets all players tracking a block entity in a server world.
+	 *
+	 * @param blockEntity the block entity
+	 * @return the players tracking the block position
+	 * @throws IllegalArgumentException if the block entity is not in a server world
+	 */
+	public static Stream<ServerPlayerEntity> tracking(BlockEntity blockEntity) {
+		Objects.requireNonNull(blockEntity, "BlockEntity cannot be null");
+
+		//noinspection ConstantConditions - IJ intrinsics don't know hasWorld == true will result in no null
+		if (!blockEntity.hasWorld() || blockEntity.getWorld().isClient()) {
+			throw new IllegalArgumentException("Only supported on server worlds!");
+		}
+
+		return tracking((ServerWorld) blockEntity.getWorld(), blockEntity.getPos());
+	}
+
+	/**
+	 * Gets all players tracking a block position in a server world.
+	 *
+	 * @param world the server world
+	 * @param pos   the block position
+	 * @return the players tracking the block position
+	 * @throws IllegalArgumentException if the world is not a server world
```

Comment:
Is this true? I only see NullPointerExceptions, also this requires an instance of ServerWorld ( or subclass )

Reply:
Those javadoc throws tags are not removed since they don't apply any more.